android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirCarePlanGeneratorTest.kt

```diff
@@ -1273,6 +1285,52 @@ class FhirCarePlanGeneratorTest : RobolectricTest() {
       }
   }
 
+  @Test
+  @ExperimentalCoroutinesApi
+  fun `Generate CarePlan should generate child immunization schedule with pre-req def and expiry timing`() =
+      runTest {
+    val planDefinitionResources =
+      loadPlanDefinitionResources("child-immunization-schedule", listOf("register-temp"))
+    val planDefinition = planDefinitionResources.planDefinition
+    val patient = planDefinitionResources.patient
+    val questionnaireResponses = planDefinitionResources.questionnaireResponses
+    val resourcesSlot = planDefinitionResources.resourcesSlot
+    fhirCarePlanGenerator.generateOrUpdateCarePlan(
+        planDefinition,
+        patient,
+        Bundle()
+          .addEntry(Bundle.BundleEntryComponent().apply { resource = patient })
+          .addEntry(
+            Bundle.BundleEntryComponent().apply { resource = questionnaireResponses.first() }
+          )
+      )!!
+      .also { println(it.encodeResourceToString()) }
+      .also { carePlan ->
+        assertCarePlan(
+          carePlan,
+          planDefinition,
+          patient,
+          patient.birthDate,
+          patient.birthDate.plusDays(4017),
+          20
+        )
+
+        resourcesSlot
+          .filter { res -> res.resourceType == ResourceType.Task }
+          .map {
+            println(it.encodeResourceToString())
+            it as Task
+          }
+          .also { tasks ->
+            assertTrue(tasks.all { it.input.firstOrNull()?.value.valueToString() == "28" })
+            assertTrue(
+              tasks.all {
+                it.restriction.period.start.asYyyyMmDd() == patient.birthDate.asYyyyMmDd()
+              }
+            )
+          }
+      }
+  }
```

Comment:
@ahsanbhatti49 I run this test and I still don't think we are doing the `partOf` addition correctly.  If you look at these dosages https://github.com/opensrp/fhircore/blob/c6a8b273ebfb2e22b32ad1b4cfdab302c674735f/android/engine/src/test/resources/plans/child-immunization-schedule/plandefinition.json#L179-L228 **Sequence 1** depends on **Sequence 0**. So the Tasks representing **Sequence 1** should have a `partOf` reference to the Tasks representing **Sequence 0**

The tasks representing sequence 0 should not have the `partOf` of timing restriction added.